### PR TITLE
fix cni issue for containerd/cni in localup

### PR DESCRIPTION
**Pull request by fisherxu**
Signed-off-by: fisherxu <fisherxu1@gmail.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
When I use localup script to install the cluste, the edgenode will get  notReady, since there's no cni netconf file in node. So it's need to create a netconf file if it doesn't exist.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
cc @gy95 @vincentgoat 
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: hack/local-up-kubeedge.sh [hunk 1]
```diff
@@ -263,6 +258,12 @@ check_control_plane_ready
 kubectl delete daemonset kindnet -nkube-system
 kubectl create ns kubeedge
 
+# install CNI plugins
+if [[ "${CONTAINER_RUNTIME}" = "remote" ]]; then
+  # we need to install CNI plugins only when we use remote(containerd) as edgecore container runtime
+  install_cni_plugins
+fi
```
Comment (fisherxu) [inline hack/local-up-kubeedge.sh:265]: Install cni when kind cluster is ready.

File: hack/lib/install.sh [hunk 2]
```diff
@@ -98,11 +98,55 @@ verify_docker_installed(){
...
+           ]
+         }
+       },
+       {
+         "type": "portmap",
+         "capabilities": {"portMappings": true}
+       }
+     ]
+    }
+EOF
```
Comment (fisherxu) [inline hack/lib/install.sh:148]: This file is copied from: https://kubernetes.io/docs/tasks/administer-cluster/migrating-from-dockershim/troubleshooting-cni-plugin-related-errors/

File: hack/lib/install.sh [hunk 3]
```diff
@@ -98,11 +98,55 @@ verify_docker_installed(){
...
+    tar Cxzvf /opt/cni/bin ${CNI_DOWNLOAD_ADDR##*/}
+    # remove installation package after the installation is complete
+    rm -rf ${KUBEEDGE_ROOT}/${CNI_DOWNLOAD_ADDR##*/}
   else
     echo "CNI plugins already installed and no need to install"
   fi
+
+  # create CNI netconf file if not exist
+  CNI_DIR="/etc/cni/net.d"
+  if [ "`ls -A $CNI_DIR`" = "" ]; then
```
Comment (gy95) [inline hack/lib/install.sh:113]: if this directory not exist, it will also NOT create netconf file, so better add a `mkdir -p /etc/cni/net.d`

Comment (gy95) [reply hack/lib/install.sh:113]: maybe just check whether file `/etc/cni/net.d/10-containerd-net.conflist` exist or not

Comment (fisherxu) [reply hack/lib/install.sh:114]: Done, when CNI_DIR not exist,  it will also print `ls: cannot access '/etc/cni/net.d': No such file or directory`

Comment (fisherxu) [reply hack/lib/install.sh:114]: Seems this file exists in node of github action, the local-up can run in CI :)

Comment (gy95) [reply hack/lib/install.sh:114]: on my local machine, this file not exists, but it seems run well 😂

File: hack/lib/install.sh [hunk 4]
```diff
@@ -98,11 +98,56 @@ verify_docker_installed(){
...
   else
     echo "CNI plugins already installed and no need to install"
   fi
+
+  # create CNI netconf file if not exist
+  CNI_DIR="/etc/cni/net.d"
+  mkdir -p /etc/cni/net.d
+  if [ "`ls -A $CNI_DIR`" = "" ]; then
+    echo -e "creating CNI netconf file..."
+    cat > /etc/cni/net.d/10-containerd-net.conflist <<EOF
```
Comment (vincentgoat) [inline hack/lib/install.sh:116]: One exceptional case is that cni0 already existed on the host before we install CNI plugins. CNI plugin works without this conf file.
Suggesting check if cni0 exists before installing: `ip a |grep cni0`

File: hack/local-up-kubeedge.sh [hunk 5]
```diff
@@ -21,6 +21,7 @@ LOG_LEVEL=${LOG_LEVEL:-2}
 TIMEOUT=${TIMEOUT:-60}s
 PROTOCOL=${PROTOCOL:-"WebSocket"}
 CONTAINER_RUNTIME=${CONTAINER_RUNTIME:-"remote"}
+CNI_DOWNLOAD_ADDR="https://github.com/containernetworking/plugins/releases/download/v1.1.1/cni-plugins-linux-amd64-v1.1.1.tgz"
```
Comment (gy95) [inline hack/local-up-kubeedge.sh:24]: moved into function install_cni_plugins, function install_cni_plugins is also used by keadm_e2e and keadm_deprecated_e2e

File: hack/lib/install.sh (hunk 3 above)
Comment (gy95) [reply hack/lib/install.sh:114]: add sudo?

**Pull request closed without merging**